app/src/main/java/org/stepik/android/view/course_list/ui/fragment/CourseListQueryFragment.kt

```diff
@@ -34,20 +54,37 @@ class CourseListActivity : FragmentActivityBase() {
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        setContentView(R.layout.activity_course_list)
         injectComponent()
 
-        val isVertical = true
-
         courseListPresenter = ViewModelProviders
             .of(this, viewModelFactory)
             .get(CourseListPresenter::class.java)
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? =
+        inflater.inflate(R.layout.fragment_course_list, container, false)
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
```

Comment:
empty states not handled

Reply:
let's pass `ViewStateDelegate<CourseListView.State>` to `CourseListViewDelegate` in order to handle empty states